src/Microsoft.Diagnostics.Monitoring.EventPipe/EventMonitor.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tracing;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.EventPipe
+{
+    /// <summary>
+    /// A stream that can monitor an event stream which is compatible with <see cref="EventPipeEventSource"/> for a specific event.
+    /// </summary>
+    internal sealed class EventMonitor : IAsyncDisposable
```

Comment:
Maybe StoppingEventMonitor? EventMonitor is very generic, especially when there are already types like EventListener.

Reply:
I'm fine with the current name or an alternate one. As long as this type stays internal a quick Ctrl-R-R can get it renamed in the future if we like.

If you want a few alternative name suggestions perhaps EventTriggerMonitor, EventMatcher, or EventScanner?